### PR TITLE
Play page

### PR DESCRIPTION
**Pull request by Plootie**
Just pushing here in case you wanna review anything will I'm working on it

### CODE REVIEW

File: SIT.Manager.Avalonia/Interfaces/IManagedProcess.cs
[the remark this answers is not in the document]
Comment (Plootie) [reply SIT.Manager.Avalonia/Interfaces/IManagedProcess.cs:1]: Thinking about it that might be a good way to go. This was just a quick change to improve readability. I was getting lost in the files 😅

File: SIT.Manager.Avalonia/Services/TarkovClientService.cs
[the remark this answers is not in the document]
Comment (Plootie) [reply SIT.Manager.Avalonia/Services/TarkovClientService.cs:1]: As i was working on it I did notice just how much overlap there was, it was just 4am and i decided to not go down that rabbit hole lol. A common class between them is probably the best way to go

**Comment by Plootie** [comment]
Sorry for taking so long to finish this up. I plan on sitting down and finishing this tomorrow

**Comment by Plootie** [comment]
I got caught up on this part for way too long, it might suck but i think it works at least? I'll work on the other parts of this tomorrow

**Comment by Plootie** [comment]
I thought this was gonna be quick to implement but uh... yeah i was wrong about that. Feel free to review what I've pushed so far if you feel like it

**Comment by Plootie** [comment]
Finally solved the big scary bug stopping me from finishing this up. Gonna swap this from a draft to full pr now but still needs testing and reviewed

File: SIT.Manager.Avalonia/ViewModels/PlayPageViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+                    CloseButtonText = "Ok",
+                }.ShowAsync();
+            }
+            else
+            {
+                using CancellationTokenSource cts = new();
+                async void serverStartEventHandler(object? sender, EventArgs e)
+                {
+                    cts.Dispose();
+                    await Dispatcher.UIThread.Invoke(async () =>
```
Comment (artehe) [inline SIT.Manager.Avalonia/ViewModels/PlayPageViewModel.cs:282]: I think InvokeAsync would be better here

**Review by artehe — COMMENTED** [review]
One general thing that I'm not a fan of is all the stuff in the Classes folder. I feel like these should be moved elsewhere but that's more of a personal stylistic thing. There's a few other things I'd want to move round with regards to where they are currently so can just do that after merging this and get it all done in one go.

Haven't tested it yet as I'm at work, but looks solid. Only found a few minor bits and pieces which I've commented on. Once those are sorted and if it still works on yours I think we can merge this and just fix any bugs that come up afterwards.

File: SIT.Manager.Avalonia/SIT.Manager.Avalonia.csproj [hunk 2]
```diff
@@ -1,28 +1,34 @@
...
+    <PackageReference Include="MegaApiClient" Version="1.10.4" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="8.0.0" />
+    <PackageReference Include="PeNet" Version="4.0.4" />
+    <PackageReference Include="SkiaSharp" Version="2.88.7" />
+    <PackageReference Include="ThemeEditor.Controls.ColorPicker" Version="11.0.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Reference Include="bsg.componentace.compression.libs.zlib">
+      <HintPath>..\..\..\..\Desktop\Tarkov\SIT\bsg.componentace.compression.libs.zlib.dll</HintPath>
```
[the remark this answers is not in the document]
Comment (Plootie) [reply SIT.Manager.Avalonia/SIT.Manager.Avalonia.csproj:31]: Ah yeah, i had the same problem with the OG launcher. This binary appears to load at least 1 other lib in the directory and is actually part of why i want to move to a self contained zlib implementation. For now i'll see what i can do to make this nice

File: SIT.Manager.Avalonia/ViewModels/PlayPageViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+            catch(Exception ex)
+            {
+                //Something BAAAAD has happened here
+                //TODO: Loggy & content dialog
+                return null;
+            }
+        }
+
+        [RelayCommand]
+        private async Task ConnectToServer()
```
[the remark this answers is not in the document]
Comment (Plootie) [reply SIT.Manager.Avalonia/ViewModels/PlayPageViewModel.cs:160]: Thanks, tbh i didn't actually know there was a difference between these two

File: SIT.Manager.Avalonia/ViewModels/PlayPageViewModel.cs (hunk 1 above)
Comment (Plootie) [reply SIT.Manager.Avalonia/ViewModels/PlayPageViewModel.cs:287]: Completely missed the fact it had an invoke async...

**Comment by artehe** [comment]
Yep all good, will merge it now 😄

**Pull request merged**